File: src/main/java/io/github/colintimbarndt/chat_emotes/data/unicode/pattern/ExactMatcher.java
package io.github.colintimbarndt.chat_emotes.data.unicode.pattern;

public record ExactMatcher(String text, String name) implements Matcher {
    @Override
    public MatchResult matchWith(String s, int offset) {
        return s.startsWith(text, offset) ? new MatchResult(text, 0, name) : null;
    }

    @Override
    public int width() {
        return 1;
    }

    @Override
    public boolean hasName() {
        return true;
    }
}

File: common_server/src/main/kotlin/io/github/colintimbarndt/chat_emotes/common/data/unicode/pattern/ExactMatcher.kt
package io.github.colintimbarndt.chat_emotes.common.data.unicode.pattern

/**
 * Matches [text] exactly
 * @see NamedExactMatcher
 */
data class ExactMatcher(val text: String, val name: String) : Matcher {
    override fun matchWith(s: String, offset: Int): MatchResult? {
        return if (s.startsWith(text, offset)) MatchResult(text, 0, name) else null
    }

    override val width = 1

    override val hasName = true
}